azurelinuxagent/common/utils/restutil.py

```diff
@@ -98,43 +146,63 @@ def _http_request(method, host, rel_uri, port=None, data=None, secure=False,
     return resp
 
 
-def http_request(method, url, data, headers=None, max_retry=3,
-                 chk_proxy=False):
-    """
-    Sending http request to server
-    On error, sleep 10 and retry max_retry times.
-    """
+def http_request(method,
+                url, data, headers=None,
+                chk_proxy=False,
+                max_retry=DEFAULT_RETRIES,
+                retry_codes=RETRY_CODES,
+                retry_delay=SHORT_DELAY_IN_SECONDS):
```

Comment:
@boumenot Makes sense, but I am, for now, preserving behavior existing as I have concerns over how changing those values will affect the "real world."